queries/djot/highlights.scm

```diff
@@ -1,14 +1,22 @@
-(heading1) @markup.heading.1
+(heading) @markup.heading
 
-(heading2) @markup.heading.2
+((heading) @markup.heading.1
+  (#match? @markup.heading.1 "^# "))
```

Comment:
Don't use `#match?` -- prefer `#eq?`, or `#lua-match?` if you must. (Would have been better to have kept the nodes in the grammar...)

Reply:
Hmm, I made this change to support arbitrary headings and I was quite pleased with how it simplified the grammar definition.. Previously we used the same setup as in markdown, but that was hardcoded to 6 heading levels.

Is it a big problem to use `#lua-match?` Something like this with `#eq?` might work too:

```
((heading
  (marker) @heading.marker) @markup.heading.1
  (#eq? @heading.marker "# "))
```

What would be the best design here in your opinion?

Reply:
As an aside, it wasn't clear to me from Neovim's help pages if we should prefer `#match?` or `#lua-match?` in our queries. Not a big deal to change of course.

Reply:
If more than six levels are useful, then I see the point ;)

If you can get away with `#eq?`, that will help performance; if not, `#lua-match?` is OK, too (but `#eq?` is even faster). `#match?` spins up the Vim regex engine, which is _really_ slow by comparison.